### PR TITLE
Mapping wizard step 4

### PR DESCRIPTION
**Pull request by michaelkro**
# Mapping Wizard Networks Step
Identical to Datastores step, but in the interest of time, only did minimal refactoring.  I did, however, change some variable naming to (hopefully) make things easier to read and follow.

Select and TreeView are extracted and shared between the datastores and networks steps

Closes https://github.com/priley86/miq_v2v_ui_plugin/issues/28

## Notes:
1. Please see commit message for 91aba12
2. Testers can `cherry-pick` 43a9a4667c7ce88d661b5 for a slightly extended data set using the following mappings (in step-2)
  * VMWare clusters 1 & 2 ---> RHV Cluster 1
  * VMWare cluster 3 ---> RHV Cluster 2

### CODE REVIEW

File: app/javascript/react/screens/App/Overview/screens/MappingWizard/components/SourceClusterSelect/SourceClusterSelect.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const mappingObject = form.includes('Networks') ? 'networks' : 'datastores';
+
+  return (
+    <div className="source-cluster-select">
+      <Form horizontal>
+        <FormGroup>
+          <Grid.Col componentClass={ControlLabel} sm={6}>
+            {__(
+              sprintf(
+                'Map source %s to target %s for cluster',
```
Comment (AparnaKarve) [inline app/javascript/react/screens/App/Overview/screens/MappingWizard/components/SourceClusterSelect/SourceClusterSelect.js]: For proper i18n string extraction, this needs to be -
`sprintf(__('Map source %s to target %s for cluster', ...)`

i.e. the `__('some text')` pattern needs to exist

File: app/javascript/react/screens/App/Overview/screens/MappingWizard/components/SourceClusterSelect/SourceClusterSelect.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                mappingObject,
+                mappingObject
```
Comment (AparnaKarve) [inline app/javascript/react/screens/App/Overview/screens/MappingWizard/components/SourceClusterSelect/SourceClusterSelect.js]: both are `mappingObject`s?

Comment (michaelkro) [reply app/javascript/react/screens/App/Overview/screens/MappingWizard/components/SourceClusterSelect/SourceClusterSelect.js]: haha, yes, I have to repeat the same string twice

**Comment by priley86** [comment]
@AparnaKarve if you have any extra time, do you mind reviewing/testing and pinging Mike R. with any issue? 

I have already started working step 5 based on this branch locally (and hoping to have that wrapped up tomorrow morning too). The Step 4 Network step behavior introduced is behaving exactly like Step 3 Datastores ( as expected). @michaelkro  is also working to construct the final POST body for `transformation_mappings` (to be integrated w/ my branch). Hopefully we can merge this along w/ my upcoming PR tomorrow.

**Comment by AparnaKarve** [comment]
Noticed an issue where the network mappings vanish in the UI (but continue to remain in the `redux form state`) when the "Back" button is pressed followed by the "Next" button.
Discussed the above issue with @michaelkro who would be fixing this issue for Step3 and 4 in a follow-up PR.
Other than the above issue and the 'Next' button not getting disabled, I think everything else seems to work

@priley86 The PR is good to go -- other issues will be fixed in a follow-up. Thanks.

**Comment by priley86** [comment]
now encapsulated in #68 ... great job w/ this @michaelkro !

maybe you'd like to add some screenshots of your work on step 4 yesterday?

**Comment by michaelkro** [comment]
I added screenshots in the comments for https://github.com/priley86/miq_v2v_ui_plugin/pull/68

@priley86 @serenamarie125 

I think I covered the main functionality, but let me know if there is anything else you'd like to see!

**Comment by michaelkro** [comment]
Closing in favor of https://github.com/priley86/miq_v2v_ui_plugin/pull/68

**Pull request closed without merging**